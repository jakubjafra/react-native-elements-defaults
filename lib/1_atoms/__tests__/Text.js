import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

const utils = require("../../utils");

describe('with no theme applied', () => {
    let OldText, Text;

    beforeAll(() => {
        let defaultStyles = {
            'atoms/text': {
            }
        };

        OldText = require('react-native-elements/src/text/Text').default;

        Text = utils.connectTheme(true, 'atoms/text', require('../Text'))(defaultStyles);
    });

    test('OldText and Text should be different', () => {
        expect(OldText).not.toBe(Text);
    });

    test('OldText renders correctly', () => {
        const tree = renderer.create(<OldText>Text</OldText>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Text renders correctly', () => {
        const tree = renderer.create(<Text>Text</Text>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldText and Text returns the same thing if no theme is applied', () => {
        const oldTree = renderer.create(<OldText>Text</OldText>).toJSON();
        const tree = renderer.create(<Text>Text</Text>).toJSON();

        expect(JSON.stringify(oldTree)).toEqual(JSON.stringify(tree));
    });
});

describe('with basic theme applied', () => {
    let OldText, Text;

    beforeAll(() => {
        let defaultStyles = {
            'atoms/text': {
                style: {
                    fontFamily: 'Arial',
                }
            }
        };

        OldText = require('react-native-elements/src/text/Text').default;

        Text = utils.connectTheme(true, 'atoms/text', require('../Text'))(defaultStyles);
    });

    test('OldText and Text should be different', () => {
        expect(OldText).not.toBe(Text);
    });

    test('OldText renders correctly', () => {
        const tree = renderer.create(<OldText>Text</OldText>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Text renders correctly', () => {
        const tree = renderer.create(<Text>Text</Text>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldText and Text should be different caused by theme applied', () => {
        const oldTree = renderer.create(<OldText>Text</OldText>).toJSON();
        const tree = renderer.create(<Text>Text</Text>).toJSON();

        expect(oldTree).not.toEqual(tree);
    });
});
