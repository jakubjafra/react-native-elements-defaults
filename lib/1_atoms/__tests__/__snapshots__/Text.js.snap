// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with basic theme applied OldText renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {},
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`with basic theme applied Text renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {},
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "fontFamily": "Arial",
      },
    ]
  }
>
  Text
</Text>
`;

exports[`with no theme applied OldText renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {},
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`with no theme applied Text renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {},
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Text
</Text>
`;
